src/rpc/core_rpc_server.cpp

```diff
@@ -3570,6 +3570,93 @@ namespace cryptonote
     return true;
   }
   //------------------------------------------------------------------------------------------------------------------------------
+  bool core_rpc_server::on_get_height_by_time(const COMMAND_RPC_GET_HEIGHT_BY_TIME::request& req, COMMAND_RPC_GET_HEIGHT_BY_TIME::response& res, epee::json_rpc::error& error_resp, const connection_context *ctx)
+  {
+    RPC_TRACKER(get_height_by_time);
+    bool r;
+    if (use_bootstrap_daemon_if_necessary<COMMAND_RPC_GET_HEIGHT_BY_TIME>(invoke_http_mode::JON_RPC, "get_height_by_time", req, res, r))
+      return r;
+
+    if((uint64_t)time(NULL) <= req.timestamp)
+    {
+      error_resp.code = CORE_RPC_ERROR_CODE_TOO_BIG_TIMESTAMP;
+      error_resp.message = "Requested timestamp is in the future";
+      return false;
+    }
+    CHECK_PAYMENT(req, res, COST_PER_HEIGHT_BY_TIME);
+
+    constexpr uint64_t buffer_time = 1 * 60 * 60; //1 hour
+    constexpr uint64_t buffer_height = 1 * 30;    //1 hour
+
+    uint64_t height_min, height_mid, height_max;
+    uint64_t timestamp_min = 0, timestamp_mid, timestamp_max = 0;
+
+    height_min = 0;
+    height_max = m_core.get_current_blockchain_height() - 1;
+
+    auto& db = m_core.get_blockchain_storage().get_db();
+
+    for (;;)
```

Comment:
I don't like to be fussy or pedantic, but I'm not too sure about these endless cycles. Would be possible to implement the same functionality without the for(;;) and break? It's very tricky to manage all the hypothetical branches imho.